File: another/Android_activity/app/src/main/java/com/example/myapplication/splashScreen.java
package com.example.myapplication;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

public class splashScreen extends Activity
{
    private long ms=0;
    private long splashTime=3000;
    private boolean splashActive = true;
    private boolean paused = false;

    @Override
    public void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activty_splash);

        Thread myThread = new Thread()
        {
            public void run()
            {
                try
                {
                    while (splashActive && ms < splashTime)
                    {
                        if(!paused)
                            ms=ms+100;
                        sleep(100);
                    }
                }catch (Exception e){}
             finally {
                    Intent intent = new Intent(splashScreen.this,MainActivity.class);
                    startActivity(intent);
                }
            }
        };
        myThread.start();

    }
}

File: app/src/main/java/com/pekeyake/covid_know/SplashScreen.kt
package com.pekeyake.covid_know

import android.content.Intent
import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.os.HandlerThread
import android.widget.RelativeLayout

class SplashScreen : AppCompatActivity()
{
    private val SPLASH_TIME_OUT = 3000L
    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)
        val background :RelativeLayout = findViewById(R.id.relativelay)
        background.setBackgroundColor(Color.DKGRAY)

        Handler().postDelayed(
            {
                val i = Intent(this@SplashScreen, HomeActivity::class.java)
                startActivity(i)
                finish()
            }, SPLASH_TIME_OUT)

    }
}